### PR TITLE
Add CI while testing with Erlang/OTP 24

### PR DESCRIPTION
**Pull request by paulo-ferraz-oliveira**
I'm opening this as draft.

I was bumping `chatterbox` in order to use with Erlang/OTP 24 and ranch 2.0.0 (and also add CI to the repo.), but running the tests makes the flow get stuck (just after `ssl:connect`). I have no time to look at this now, but will be leaving it here in case anybody wants to take a jab at it, or guide guidance as to what might be making the tests get stuck.

### CODE REVIEW

File: rebar.config [hunk 1]
```diff
@@ -24,7 +22,19 @@
             nowarn_export_all
            ]},
           {deps,
-           [{ranch, "1.2.1"}]}
+           [{ranch, "2.0.0"}, {cth_readable, "v1.5.1"}]},
```
Comment (paulo-ferraz-oliveira) [inline rebar.config:25]: `cth_readable` added for `dialyzer`ing purposes

File: rebar.lock [hunk 2]
```diff
@@ -1,6 +1,8 @@
-{"1.1.0",
+{"1.2.0",
```
Comment (paulo-ferraz-oliveira) [inline rebar.lock:1]: This is rebar 3.16.1

File: rebar.config [hunk 3]
```diff
@@ -13,8 +13,6 @@
 
 {ct_opts, [{verbose,true}]}.
 
-{dialyzer, [unknown]}.
```
Comment (paulo-ferraz-oliveira) [inline rebar.config:16]: Moved to the `test` profile so I could take advantage of `deps` there.

**Comment by paulo-ferraz-oliveira** [comment]
Looking at open pull requests I see that @tsloughter's already attempted at some of the changes I've made here (unfortunately I didn't look for open requests sooner), but they remain unmerged (like going from Travis to GHA). I can wait on whatever is done in other pull requests to check if anything remains open in this one.

**Comment by tsloughter** [comment]
Could you open against my fork instead? I have permissions on this repo too but have been wanting to do more cleanup on my code before merging into this repo. I need to go ahead and do it soon... so you could wait on that instead of opening on my fork.

**Comment by paulo-ferraz-oliveira** [comment]
Sure. Getting different errors, there, which might even be a good thing.

**Comment by paulo-ferraz-oliveira** [comment]
Closed in favor of https://github.com/tsloughter/chatterbox/pull/8.

**Pull request closed without merging**